addons/escoria-core/game/main.gd

```diff
@@ -1,43 +1,57 @@
+# Escoria main room handling and scene switcher
 extends Node
 
 # This script is basically the scene-switcher.
 
 # Global id of the last scene the player was before current scene
-var last_scene_global_id
+var last_scene_global_id: String
+
 # Current scene room being displayed
-var current_scene
+var current_scene: Node
 
+# The Escoria context currently in wait state
 var wait_level
 
+# FIXME Document this variable
 var screen_ofs = Vector2(0, 0)
 
-# ESCBackgroundMusic node
+# Reference to the ESCBackgroundMusic node
 onready var bg_music = $bg_music
+
+# Reference to the scene transition node
 onready var scene_transition = $layers/curtain/scene_transition
 
 
-# Set the new current scene
+# Connect the wait timer event
+func _ready() -> void:
+	$layers/wait_timer.connect("timeout", self, "_on_wait_finished")
+
+
+# Set current scene
 #
 # #### Parameters
 #
-# - p_scene: Current scene to set
-func set_scene(p_scene: Node):
+# - p_scene: Scene to set
+# - run_events: Play the events in the setup node
+# - *Returns*: A dictionary of ESC events
+func set_scene(p_scene: Node, run_events=true) -> void:
```

Comment:
It seems `run_events` was added here but not used.

Reply:
Good catch